tests/test_ssl_transport_close.py

```diff
@@ -0,0 +1,140 @@
+import asyncio
+import platform
+import ssl
+import sys
+import warnings
+from typing import Any, Tuple, cast
+
+import pytest
+
+from aiohttp import TCPConnector, web
+from aiohttp.client import ServerDisconnectedError
+from aiohttp.pytest_plugin import AiohttpClient, AiohttpServer
+from aiohttp.test_utils import TestClient, TestServer
+
+
+async def handle_root(request: web.Request) -> web.Response:
+    """respond after server was restarted, forcing connection_lost"""
+    cq = request.app["cq"]
+    dq = request.app["dq"]
+
+    await dq.put(0)
+
+    await cq.get()
+    cq.task_done()
+
+    return web.Response(text="")
+
+
+async def _prepare(
+    aiohttp_server: Any,
+    aiohttp_client: Any,
+    ssl_ctx: ssl.SSLContext,
+    client_ssl_ctx: ssl.SSLContext,
+    cq: "asyncio.Queue[int]",
+    dq: "asyncio.Queue[int]",
+) -> Tuple[TestServer, TestClient]:
+    app = web.Application()
+    app["cq"] = cq
+    app["dq"] = dq
+    app.router.add_get("/", handle_root)
+    server = await aiohttp_server(app, port=0, ssl=ssl_ctx)
+
+    session = await aiohttp_client(server, connector=TCPConnector(ssl=client_ssl_ctx))
+
+    # replace SockSite … it is different and blocks
+    await dq.put(0)
+    await _restart(server.runner, ssl_ctx, session.port, cq, dq)
+    await cq.get()
+
+    return server, session
+
+
+async def _restart(
+    runner: web.BaseRunner,
+    ssl_ctx: ssl.SSLContext,
+    port: int,
+    cq: "asyncio.Queue[int]",
+    dq: "asyncio.Queue[int]",
+) -> None:
+    """restart service to force connection_lost"""
+    await dq.get()
+    dq.task_done()
+    site = next(iter(runner.sites))
+    await site.stop()
```

Comment:
In 3.12 it hangs here - you do not get 

> DEBUG    aiohttp.server:web_protocol.py:434 Ignored premature client disconnection

as you get in 3.11

Reply:
I don't see anything suspicious in there, unless the asyncio server is not closing:
https://github.com/commonism/aiohttp/blob/ebe2a48c282f2484ee1766af5c4d52ff3431b43a/aiohttp/web_runner.py#L83

Everything else looks like it has a 60s timeout, so would fail before the 5 min timeout we're seeing.

Reply:
If it is the asyncio server not closing, then we should get a bug filed for the regression.